community/kernel/src/main/java/org/neo4j/kernel/impl/api/DataIntegrityValidatingStatementOperations.java

```diff
@@ -204,7 +221,7 @@ public void constraintDrop( KernelStatement state, ConstraintDescriptor descript
         schemaWriteDelegate.constraintDrop( state, descriptor );
     }
 
-    private void assertIndexDoesNotExist( KernelStatement state, OperationContext context,
+    private void assertUniqueIndexDoesNotExist( KernelStatement state, OperationContext context,
```

Comment:
This method will throw regardless of whether the existing index is used for uniqueness constraints or not, won't it?